WChat/src/app/redux/user/reducer.js

```diff
@@ -38,17 +43,40 @@ const user = (state = initialState, action) => {
         chatsLoading: false
       })
     case actionTypes.GET_CHATS_SUCCESS:
-      const oldContactIndex = state.contacts.findIndex((contact) => contact.id === action.receiverId);
-      const oldContact = state.contacts[oldContactIndex];
-      let newContact =  {
+      if(!action.chats.length){
+        return Immutable.merge(state, {
+          currentChat: []
+        })
+      }
+      oldContactIndex = state.contacts.findIndex((contact) => contact.id === action.receiverId);
+      oldContact = state.contacts[oldContactIndex];
+      newContact =  {
         ...oldContact,
         chats: action.chats,
         lastChat: action.chats.reduce((latest, current) => latest.createdAt > current.createdAt ? latest : current)
       }
-      let newContacts = cloneAndInsertInArray(state.contacts, newContact, oldContactIndex)
-      
+      newContacts = cloneAndInsertInArray(state.contacts, newContact, oldContactIndex)
+      return Immutable.merge(state, {
+        contacts: newContacts,
+        currentChat: action.chats.reverse()
+      })
+    case actionTypes.SEND_MESSAGE_FAILURE:
+      return Immutable.merge(state, {
+        error: action.error
+      })
+    case actionTypes.MESSAGE_SENT:
+      oldContactIndex = state.contacts.findIndex((contact) => contact.id === action.receiverId);
+      oldContact = state.contacts[oldContactIndex];
+      newChats = [...oldContact.chats, action.message];
+      newContact = {
+        ...oldContact,
+        chats: newChats,
+        lastChat: action.message
+      }
+      newContacts = cloneAndInsertInArray(state.contacts, newContact, oldContactIndex);
```

Comment:
Extract all the logic to a function, so you dont have to repeat the code everytime you need to update the contacts